recipes/libbigwig/all/conanfile.py

```diff
@@ -0,0 +1,92 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import collect_libs, get
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.50.0"
+
+
+class LibBigWigConan(ConanFile):
+    name = "libbigwig"
+    description = "A C library for handling bigWig files"
+    topics = ("bioinformatics", "bigwig", "bigbed")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/dpryan79/libBigWig"
+    license = "MIT"
+    settings = "arch", "build_type", "compiler", "os"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_curl": [True, False],
+        "with_zlibng": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_curl": True,
+        "with_zlibng": False
+    }
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.variables["ENABLE_TESTING"] = False
+        tc.variables["WITH_CURL"] = self.options.with_curl
+        tc.variables["WITH_ZLIBNG"] = self.options.with_zlibng
+
+        tc.generate()
+
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
+
+    def requirements(self):
+        if self.options.with_curl:
+            self.requires("libcurl/[>=7.83 <7.85]")
+        if self.options.with_zlibng:
+            self.requires("zlib-ng/2.0.6")
+        else:
+            self.requires("zlib/1.2.12")
```

Comment:
Most often we have `with_compression: ["zlib", "zlib-ng"]` when there's an or option 

Its more explicit when there's a choice for the user

(just a suggestion)

Reply:
Yes, it seems be much better, at least it's much more explicit about the dependency used.